src/rulesets/Base/fastmath_able.jl

```diff
@@ -133,15 +133,15 @@ let
 
         @scalar_rule x + y (One(), One())
         @scalar_rule x - y (One(), -1)
-        @scalar_rule x / y (inv(y), -((x / y) / y))
+        @scalar_rule x / y (one(Ω) / y, -(Ω / y))
         #log(complex(x)) is required so it gives correct complex answer for x<0
         @scalar_rule(x ^ y,
             (ifelse(iszero(x), zero(Ω), y * Ω / x), Ω * log(complex(x))),
         )
         # x^y for x < 0 errors when y is not an integer, but then derivative wrt y
         # is undefined, so we adopt subgradient convention and set derivative to 0.
         @scalar_rule(x::Real ^ y::Real,
-            (ifelse(iszero(x), zero(Ω), y * Ω / x), Ω * log(ifelse(x ≤ 0, one(x), x))),
+            (ifelse(iszero(x), zero(Ω), y * Ω / x), Ω * log(convert(typeof(Ω), ifelse(x ≤ 0, one(x), x)))),
```

Comment:
I did this because otherwise for integer x and float32  y I was obtaining
```julia
  Expression: frule((Zero(), Δx, Δy), f, 2, y) isa Tuple{T, T}
   Evaluated: (1.2687975f0, 0.44214729389061513) isa Tuple{Float32,Float32}
```
I'm not sure convert is the best way to do it or if we want to care about this case at all. If we don't I'll just remove this change and the corresponding test

Reply:
Good catch. This approach looks right. Perhaps this is cleaner?
```suggestion
            (ifelse(iszero(x), zero(Ω), y * Ω / x), Ω * log(oftype(Ω, ifelse(x ≤ 0, one(x), x)))),
```